### PR TITLE
[LaTeX] Inherit LaTeX from TeX

### PR DESCRIPTION
**Pull request by deathaxe**
This PR is an attempt to avoid (future) code duplication when improving TeX/LaTeX.

It extends LaTeX using TeX as base syntax.

I am uncertain whether test coverage is high enough to dedect possible regressions. Hence someone with good LaTeX experience should have a look at it.

### CODE REVIEW

**Comment by deathaxe** [comment]
v2 syntax requires multiple meta scope adjustments and we probably would want to replace all `pop: true` by `pop: 1` then.

That's basically unrelated with just inheriting LaTeX from TeX and would cause trouble (avoidable conflicts) with @ngc92's other PRs.

This WIP PR is more or less meant as inspiration for him to learn how inheritance works and how it hopefully could help avoiding code duplications when reworking/improving LaTeX.

**Comment by ngc92** [comment]
I took a quick look at this. Do we need to have `macro-braces` as part of LaTeX? Isn't it the same as in TeX?

**Comment by deathaxe** [comment]
Maybe. This PR is just a starting point. Further work may be needed or useful.

File: Haskell/Literate Haskell.sublime-syntax [hunk 1]
```diff
@@ -10,7 +10,7 @@ file_extensions:
   - lhs
 
 contexts:
-  plain-tex:
+  begin-end-commands:
```
[the remark this answers is not in the document]
Comment (deathaxe) [reply Haskell/Literate Haskell.sublime-syntax:13]: It is related as this PR removes `plain-tex` context and thus would break Literate Haskell.

**Comment by deathaxe** [comment]
With regards to https://github.com/deathaxe/sublime-packages/pull/1#issuecomment-1320851441 I've added overrides for braced groups for illustration purpose.

Please note, how we now will see various `...brace.end.latex` in expressions re-used from `TeX` - the vice versa of your concern.

So basically using the base syntax's final scope `.tex` is probably a better and more generic choice.

**Comment by ngc92** [comment]
then I think going with `.tex` for all braces makes sense. 

In any case, what's the plan going forward with this PR. For further changes, it would probably be good to get the inheritance set up soon, so that other improvements can build on it.

**Comment by deathaxe** [comment]
Added a couple more commits to give some inspiration of possible code structure for supporting inheritance the best we can - just some experiences from other syntaxes.

In general I think of creating a `develop/latex` branch in sublimehq/Packages repo we could merge smaller PRs into and keep going without touching master as it seems to become some kind of "Rewrite LaTeX" project ... just to reduce noice on master.

Once a final state is achieved we could merge everything into master at once.

Thoughts?

**Comment by ngc92** [comment]
sounds good to me. I mainly want to avoid the situation where the changes I'm doing now need to be re-done for the inheritance later on

**Comment by deathaxe** [comment]
Sure. This PR should just have been an inspiration for you mainly. I'll try to git this content into a develop branch which you can base on furhter work.

**Comment by deathaxe** [comment]
Closing this in favor of the general development branch/PR #3575.

**Pull request closed without merging**